test/model/UserSpec.scala

```diff
@@ -49,4 +54,49 @@ class UserSpec extends PlaySpec {
           |}""".stripMargin.replace("\n", "")
     }
   }
+
+  "fromOktaUser" should {
+
+    "create a User from OktaUser when all fields are present and valid" in {
+      val additionalProperties = Map("legacyIdentityId" -> "87654321").asJava
+      val profile = mock[UserProfile]
+      when(profile.getEmail).thenReturn("test@example.com")
+      when(profile.getAdditionalProperties).thenReturn(additionalProperties)
+      val oktaUser = mock[OktaUser]
+      when(oktaUser.getId).thenReturn("12345678")
+      when(oktaUser.getProfile).thenReturn(profile)
+      val user = User.fromOktaUser(oktaUser)
+      user.oktaId shouldBe "12345678"
+      user.emailAddress shouldBe Some("test@example.com")
+      user.legacyIdentityId shouldBe "87654321"
+    }
+
+    "create a User from OktaUser when some fields are missing" in {
+      val additionalProperties = Map("legacyIdentityId" -> "87654321").asJava
+      val profile = mock[UserProfile]
+      when(profile.getEmail).thenReturn(null)
+      when(profile.getAdditionalProperties).thenReturn(additionalProperties)
+      val oktaUser = mock[OktaUser]
+      when(oktaUser.getId).thenReturn("12345678")
+      when(oktaUser.getProfile).thenReturn(profile)
+      val user = User.fromOktaUser(oktaUser)
+      user.oktaId shouldBe "12345678"
+      user.emailAddress shouldBe None
+      user.legacyIdentityId shouldBe "87654321"
+    }
+
+    "create a User from OktaUser when some fields are present but have invalid values" in {
+      val additionalProperties = Map("legacyIdentityId" -> "87654321").asJava
+      val profile = mock[UserProfile]
+      when(profile.getEmail).thenReturn(" ")
+      when(profile.getAdditionalProperties).thenReturn(additionalProperties)
+      val oktaUser = mock[OktaUser]
+      when(oktaUser.getId).thenReturn("12345678")
+      when(oktaUser.getProfile).thenReturn(profile)
+      val user = User.fromOktaUser(oktaUser)
+      user.oktaId shouldBe "12345678"
+      user.emailAddress shouldBe None
+      user.legacyIdentityId shouldBe "87654321"
+    }
+  }
```

Comment:
in the future we could also add the integration tests in `CODE` like we started adding in IDAPI.